lib/DelegatedModule.js

```diff
@@ -2,79 +2,80 @@
+const Module = require("./Module");
+const OriginalSource = require("webpack-sources").OriginalSource;
+const RawSource = require("webpack-sources").RawSource;
+const WebpackMissingModule = require("./dependencies/WebpackMissingModule");
+const DelegatedSourceDependency = require("./dependencies/DelegatedSourceDependency");
 
-DelegatedModule.prototype = Object.create(Module.prototype);
-DelegatedModule.prototype.constructor = DelegatedModule;
+class DelegatedModule {
+	constructor(sourceRequest, data, type, userRequest) {
+		Module.call(this);
+		this.sourceRequest = sourceRequest;
+		this.request = data.id;
+		this.meta = data.meta;
+		this.type = type;
+		this.userRequest = userRequest;
+		this.built = false;
+		this.delegateData = data;
+	}
 
-DelegatedModule.prototype.delegated = true;
+	identifier() {
+		return "delegated " + JSON.stringify(this.request) + " from " + this.sourceRequest;
+	}
 
-DelegatedModule.prototype.identifier = function() {
-	return "delegated " + JSON.stringify(this.request) + " from " + this.sourceRequest;
-};
+	readableIdentifier() {
+		return "delegated " + this.userRequest + " from " + this.sourceRequest;
+	}
+
+	needRebuild() {
+		return false;
+	}
 
-DelegatedModule.prototype.readableIdentifier = function() {
-	return "delegated " + this.userRequest + " from " + this.sourceRequest;
-};
+	build(options, compilation, resolver, fs, callback) {
+		this.built = true;
+		this.builtTime = new Date().getTime();
+		this.usedExports = true;
+		this.providedExports = this.delegateData.exports || true;
+		this.dependencies.length = 0;
+		this.addDependency(new DelegatedSourceDependency(this.sourceRequest));
+		callback();
+	}
 
-DelegatedModule.prototype.needRebuild = function() {
-	return false;
-};
+	unbuild() {
+		this.built = false;
+		Module.prototype.unbuild.call(this);
+	}
 
-DelegatedModule.prototype.build = function(options, compilation, resolver, fs, callback) {
-	this.built = true;
-	this.builtTime = new Date().getTime();
-	this.usedExports = true;
-	this.providedExports = this.delegateData.exports || true;
-	this.dependencies.length = 0;
-	this.addDependency(new DelegatedSourceDependency(this.sourceRequest));
-	callback();
-};
+	source() {
+		const sourceModule = this.dependencies[0].module;
+		let str;
 
-DelegatedModule.prototype.unbuild = function() {
-	this.built = false;
-	Module.prototype.unbuild.call(this);
-};
+		if(!sourceModule) {
+			str = WebpackMissingModule.moduleCode(this.sourceRequest);
+		} else {
+			switch(this.type) {
+				case "require":
+					str += "(" + JSON.stringify(this.request) + ");";
+					break;
+				case "object":
+					str += "[" + JSON.stringify(this.request) + "];";
+					break;
+			}
+		}
 
-DelegatedModule.prototype.source = function() {
-	var sourceModule = this.dependencies[0].module;
-	var str;
-	if(!sourceModule) {
-		str = WebpackMissingModule.moduleCode(this.sourceRequest);
-	} else {
-		str = "module.exports = (__webpack_require__(" + sourceModule.id + "))";
```

Comment:
This line is missing in the rewrite; the `str +=` lines will append to `undefined`.